core/src/main/java/tc/oc/pgm/fireworks/PostMatchFireworkMatchModule.java

```diff
@@ -0,0 +1,109 @@
+package tc.oc.pgm.fireworks;
+
+import com.google.common.collect.ImmutableList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.bukkit.Color;
+import org.bukkit.FireworkEffect;
+import org.bukkit.FireworkEffect.Type;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.scheduler.BukkitTask;
+import tc.oc.pgm.Config;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.match.MatchModule;
+import tc.oc.pgm.api.match.MatchScope;
+import tc.oc.pgm.api.match.event.MatchFinishEvent;
+import tc.oc.pgm.api.party.Competitor;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.events.ListenerScope;
+import tc.oc.util.bukkit.BukkitUtils;
+
+@ListenerScope(MatchScope.LOADED)
+public class PostMatchFireworkMatchModule implements MatchModule, Listener {
+
+  private BukkitTask task;
```

Comment:
Don't use `BukkitTask`, instead `Match#getScheduler(MatchScope.LOADED)`

Reply:
Seems like you do already, so no need to store the `BukkitTask`, it will get canceled automatically.